porcupine/plugins/hover.py

```diff
@@ -16,24 +16,27 @@ class Response(utils.EventDataclass):
 class HoverManager:
     def __init__(self, textwidget: tkinter.Text):
         self._textwidget = textwidget
-        self._label: tkinter.Label | None = None
-        self._location = textwidget.index("insert")
-
-    def hide_label(self, junk: object = None) -> None:
-        if self._label is not None:
-            self._label.destroy()
-            self._label = None
-
-    def _show_label(self, message: str) -> None:
-        self.hide_label()
         self._label = tkinter.Label(
             self._textwidget,
-            text=message,
             justify="left",
             # some hack to get a fitting popup bg
             bg=utils.mix_colors(self._textwidget["bg"], self._textwidget["fg"], 0.8),
             fg=self._textwidget["fg"],
         )
+        self._location = textwidget.index("insert")
+
+    def hide_label(self, junk: object = None) -> None:
+        self._label.place_forget()
+
+    def _show_label(self, message: str) -> None:
+        self.hide_label()
+
+        if message != self._label["text"]:
```

Comment:
Maybe we should get everything under the same `if` statement? If I hover the `i` of `list` in Python code and I move my mouse over the `s`, I currently get a newly placed popup even though the text doesn't change.

Reply:
`_request_hover` hides the label first anyways, and moving the `after` essentially makes this `if` redundant.